common/src/main/java/feast/common/models/FeatureSetReference.java

```diff
@@ -48,6 +48,20 @@ public static FeatureSetReference of(String projectName, String featureSetName)
     return FeatureSetReference.of(projectName, featureSetName, -1);
   }
 
+  public static FeatureSetReference parse(String reference) {
```

Comment:
Can we add a comment here please, since this is a very important part of the code.